tests/unittests/ExecutorTest.cpp

```diff
@@ -201,16 +201,20 @@ class ExecutorTest final {
 public:
   /// Constructor.
   ExecutorTest(const std::shared_ptr<Executor> &executor,
-               std::unique_ptr<DAGNode> root, std::unique_ptr<Type> type,
-               DAGNodeNameMapTy nodes, PlaceholderNameMapTy placeholders,
+               std::unique_ptr<DAGNode> root, std::unique_ptr<Module> module,
+               std::unique_ptr<Type> type, DAGNodeNameMapTy nodes,
+               PlaceholderNameMapTy placeholders,
                std::unique_ptr<ExecutionContext> inputContext,
                std::unique_ptr<ExecutionContext> outputContext,
                RunIdentifierTy runId, bool expectSuccess)
-      : executor_(executor), root_(std::move(root)), type_(std::move(type)),
-        nodes_(std::move(nodes)), placeholders_(std::move(placeholders)),
+      : executor_(executor), root_(std::move(root)), module_(std::move(module)),
+        type_(std::move(type)), nodes_(std::move(nodes)),
+        placeholders_(std::move(placeholders)),
         inputContext_(std::move(inputContext)),
         outputContext_(std::move(outputContext)), runId_(runId),
-        expectSuccess_(expectSuccess), testRun_(false) {}
+        expectSuccess_(expectSuccess), testRun_(false) {
+    root_->module = module_.get();
```

Comment:
I would personally prefer that this be done in `ExecutorTestBuilder`.

Reply:
I was getting a segfault in the concurrent test when I added it to the builder. Moving it to the test seems to behave properly.